### ISSUE TITLE
Executing OrleansCounterControl.exe from Microsoft.Orleans.CounterControl version 1.1.2 nuget fails  

### ISSUE THREAD

**Issue by patrickac** [dotnet/orleans#1547]
Executing the OrleansCounterControl.exe results in following error:

C:\Source\Repos\STS\Services\Commerce\out\Wallet.Grains.Tests>OrleansCounterControl.exe
OrleansCounterControl v1.0.0.0
Registering Orleans performance counters with Windows
Error registering Orleans counters - {0}System.IO.FileLoadException: Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToke
n=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at Orleans.CodeGenerator.RoslynCodeGenerator.GenerateAndLoadForAssemblies(Assembly[] inputs)
   at Orleans.CodeGeneration.CodeGeneratorManager.GenerateAndCacheCodeForAllAssemblies()
   at Orleans.Runtime.AssemblyProcessor.Initialize()
   at Orleans.Runtime.AssemblyLoader.NewAssemblyLoader(Dictionary`2 dirEnumArgs, IEnumerable`1 pathNameCriteria, IEnumerable`1 reflectionCriteria, TraceLogger logger)
   at Orleans.Runtime.AssemblyLoader.LoadAssemblies(Dictionary`2 dirEnumArgs, IEnumerable`1 pathNameCriteria, IEnumerable`1 reflectionCriteria, TraceLogger logger)
   at Orleans.Runtime.SiloAssemblyLoader.LoadApplicationAssemblies()
   at Orleans.Runtime.GrainTypeManager.Start(Boolean strict)
   at Orleans.Counter.Control.CounterControl.RegisterWindowsPerfCounters(Boolean useBruteForce)

Assembly manager loaded from:  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\clr.dll
Running under executable  C:\Source\Repos\STS\Services\Commerce\out\Wallet.Grains.Tests\OrleansCounterControl.exe
--- A detailed error log follows.

=== Pre-bind state information ===
LOG: DisplayName = Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
(Fully-specified)
LOG: Appbase = file:///C:/Source/Repos/STS/Services/Commerce/out/Wallet.Grains.Tests/
LOG: Initial PrivatePath = NULL
Calling assembly : OrleansCodeGenerator, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null.
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Source\Repos\STS\Services\Commerce\out\Wallet.Grains.Tests\OrleansCounterControl.exe.Config
LOG: Using host configuration file:
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
LOG: Attempting download of new URL file:///C:/Source/Repos/STS/Services/Commerce/out/Wallet.Grains.Tests/Microsoft.CodeAnalysis.CSharp.DLL.
WRN: Comparing the assembly name resulted in the mismatch: Minor Version
ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.

Error running OrleansCounterControl.exe
Exception = System.IO.FileLoadException: Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one o
f its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at Orleans.CodeGenerator.RoslynCodeGenerator.GenerateAndLoadForAssemblies(Assembly[] inputs)
   at Orleans.CodeGeneration.CodeGeneratorManager.GenerateAndCacheCodeForAllAssemblies()
   at Orleans.Runtime.AssemblyProcessor.Initialize()
   at Orleans.Runtime.AssemblyLoader.NewAssemblyLoader(Dictionary`2 dirEnumArgs, IEnumerable`1 pathNameCriteria, IEnumerable`1 reflectionCriteria, TraceLogger logger)
   at Orleans.Runtime.AssemblyLoader.LoadAssemblies(Dictionary`2 dirEnumArgs, IEnumerable`1 pathNameCriteria, IEnumerable`1 reflectionCriteria, TraceLogger logger)
   at Orleans.Runtime.SiloAssemblyLoader.LoadApplicationAssemblies()
   at Orleans.Runtime.GrainTypeManager.Start(Boolean strict)
   at Orleans.Counter.Control.CounterControl.RegisterWindowsPerfCounters(Boolean useBruteForce)
   at Orleans.Counter.Control.CounterControl.Run()

Assembly manager loaded from:  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\clr.dll
Running under executable  C:\Source\Repos\STS\Services\Commerce\out\Wallet.Grains.Tests\OrleansCounterControl.exe
--- A detailed error log follows.

=== Pre-bind state information ===
LOG: DisplayName = Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
(Fully-specified)
LOG: Appbase = file:///C:/Source/Repos/STS/Services/Commerce/out/Wallet.Grains.Tests/
LOG: Initial PrivatePath = NULL
Calling assembly : OrleansCodeGenerator, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null.
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Source\Repos\STS\Services\Commerce\out\Wallet.Grains.Tests\OrleansCounterControl.exe.Config
LOG: Using host configuration file:
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework64\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
LOG: Attempting download of new URL file:///C:/Source/Repos/STS/Services/Commerce/out/Wallet.Grains.Tests/Microsoft.CodeAnalysis.CSharp.DLL.
WRN: Comparing the assembly name resulted in the mismatch: Minor Version
ERR: Failed to complete setup of assembly (hr = 0x80131040). Probing terminated.

**Comment by patrickac** [comment]
Determined to be erroneous

**Comment by Andriy06** [comment]
same problem here:

```
OrleansCounterControl v1.2.2.0

Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

Sorry I didn't get what you mean by "Determined to be erroneous"

**Comment by sergeybykov** [comment]
@Andriy06 Do you have `'Microsoft.CodeAnalysis.CSharp.dll` and its dependencies in the folder from where you run `OrleansCounterControl.exe`?

**Comment by Andriy06** [comment]
I run OrleansCounterControl.exe from my Orleans Silo project (generated by Visual Studio extension) bin/Release output folder. I tried to analyse its dependencies but `Microsoft.CodeAnalysis.CSharp.dll` isn't among them. As I specified above I do have `Microsoft.CodeAnalysis.CSharp.dll` in this folder, v.1.2.2. I managed to execute OrleansCounterControl.exe by hand picking dependencies in a separate folder, Microsoft.CodeAnalysis.CSharp.dll v.1.0.0, so seems it was a problem. Does it really depends on this particular version or is it just too restrictive configuration? Would be great to be able to run latest version of OrleansCounterControl with latest stable version of Microsoft.CodeAnalysis.CSharp.dll.

**Comment by sergeybykov** [comment]
https://github.com/dotnet/orleans/blob/master/src/NuGet/Microsoft.Orleans.CounterControl.nuspec explicitly lists `Microsoft.CodeAnalysis.CSharp 1.0.0` as its dependency, which brings the rest of them.

**Comment by rosdi** [comment]
I am following the tutorial here http://dotnet.github.io/orleans/Tutorials/Running-in-a-Stand-alone-Silo.html, all was fine until I upgraded all nuget packages in all 3 projects to the latest.

`System.IO.FileLoadException: Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=1.0.0.0...`

I had to revert back my nuget packages, any reason why this is Closed?

**Comment by sergeybykov** [comment]
Try upgrading to 1.3.0-beta1.

**Comment by jdom** [comment]
Yes, I believe it is specific to version 1.0.0.0. If for some reason you must also update the code analysis library (not required by Orleans), then you might try explicitly adding binding redirects to OrleansCounterControl.exe.config as nuget will not do it for you

**Comment by rosdi** [comment]
Ah didn't know there is a beta version available... will try again later.. thanks.

**Comment by rosdi** [comment]
Hi @sergeybykov , just want to update on this issue... even on 1.3.0-beta1 still specifically requires Microsoft.CodeAnalysis.CSharp Version=1.0.0.0 

I could try a binding redirect as suggested by jdom but there is no need... version 1.0.0.0 works fine for now.

**Comment by sergeybykov** [comment]
@rosdi We just published 1.3.0-beta2 where we bumped up versions of the Roslyn dependency NuGets. Can you check if that solves your problem?
